docs/design/proposals/rbd-volume-healer.md

```diff
@@ -0,0 +1,94 @@
+# RBD NBD VOLUME HEALER
+
+- [RBD NBD VOLUME HEALER](#rbd-nbd-volume-healer)
+  - [Rbd Nbd](#rbd-nbd)
+    - [Advantages of userspace mounters](#advantages-of-userspace-mounters)
+    - [Side effects of userspace mounters](#side-effects-of-userspace-mounters)
+  - [Volume Healer](#volume-healer)
+    - [More thoughts](#more-thoughts)
+
+## Rbd Nbd
+
+The RBD CSI plugin will provision new RBD images and attach and mount those
+to workloads. Currently, the default mounter is krbd, which uses the kernel
+rbd driver to mount the rbd images onto the application node. Here on
+at cephcsi we will also have a userspace way of mounting the RBD images,
+via RBD-NBD.
+
+[Rbd-nbd](https://docs.ceph.com/en/latest/man/8/rbd-nbd/) is a client for
+RADOS block device (rbd) images like the existing rbd kernel module. It
+will map an rbd image to an NBD (Network Block Device) device, allowing
+access to it as a regular local block device.
+
+![csi-rbdnbd](https://user-images.githubusercontent.com/12432241/125400940-23e75d00-e3d0-11eb-806c-bab09318de54.png)
+
+It’s worth making a note that the rbd-nbd processes will run on the
+client-side, which is inside the `csi-rbdplugin` node plugin.
+
+### Advantages of userspace mounters
+
+- Adding changes to client logic to accommodate new features will be easy
+- Container upgrades will be independent of the host node, which means if
+  there any changes in the client logic, we don’t have to reboot the node
+  as the changes will be shipped inside the container.
+- We will be a better citizen in K8s by switching to the userspace model.
+- Unlike krbd, rbd-nbd uses librbd user-space library that gets most of the
+  development focus, and hence rbd-nbd will be future rich.
```

Comment:
**will be _more feature_ rich** (not future)

Reply:
oops! ok